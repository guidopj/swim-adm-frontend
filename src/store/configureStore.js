import { createStore, applyMiddleware, compose } from 'redux';
import createSagaMiddleware from 'redux-saga';
import redirectMiddleware from 'middlewares/redirectMiddleware'
import history from 'history.js'
import { routerMiddleware } from 'connected-react-router'
import { persistStore, persistReducer } from 'redux-persist'
import storage from 'redux-persist/lib/storage'

import rootSaga from 'sagas';
import rootReducer from 'reducers/rootReducer';

const persistConfig = {
	key: 'root',
	storage,
  }

  const mainReducer = rootReducer(history)
  const persistedReducer = persistReducer(persistConfig, mainReducer)

/** Saga Middleware */
const sagaMiddleware = createSagaMiddleware();

let middlewares = applyMiddleware(sagaMiddleware, routerMiddleware(history), redirectMiddleware);


export default function configureStore (initialState) {
	const composeEnhancers = window.__REDUX_DEVTOOLS_EXTENSION_COMPOSE__ || compose;
    const store = createStore(persistedReducer,initialState, composeEnhancers(
        middlewares,
	  )
	);
	let persistor = persistStore(store)
	
	sagaMiddleware.run(rootSaga);
    return { store, persistor }
}

/* export const sagaMiddleware = createSagaMiddleware()

const composeStore = compose(
	applyMiddleware(
		redirectMiddleware,
		metricsMiddleware,
		sagaMiddleware,
		generalMiddleware,
		postMessageMiddleware
	),
	window.devToolsExtension ? window.devToolsExtension() : f => f
)(createStore)

const store = composeStore(reducers)

sagaMiddleware.run(rootSaga)

export default store */