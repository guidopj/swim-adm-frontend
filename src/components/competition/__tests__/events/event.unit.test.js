import React from 'react';
import Event from 'components/competition/events/event';
import renderer from 'react-test-renderer';
import history from 'history.js'
import { ConnectedRouter } from 'connected-react-router'
import { Provider } from 'react-redux'
import configureStore from 'store/configureStore'


describe ('Events with state and props', () => {

    const events = [
        {
            eventNro: 1,
            meters: 50,
            style: "BACKSTROKE",
            genre: "MALE",
            category_from_age: 18,
            category_to_age: 23
        },
        {
            eventNro: 2,
            meters: 50,
            style: "BACKSTROKE",
            genre: "MALE",
            category_from_age: 18,
            category_to_age: 23
        },
        {
            eventNro: 3,
            meters: 50,
            style: "BACKSTROKE",
            genre: "MALE",
            category_from_age: 18,
            category_to_age: 23
        },
    ]

const { store } = configureStore({
    competitions: {}
  })

  it('renders correctly', () => {
    const component = renderer.create(
        <Provider store={store}>
            <ConnectedRouter history={history}>
                <Event
                    events={events}
                />
            </ConnectedRouter>
        </Provider>
    )
    const tree = component.toJSON();
    expect(tree).toMatchSnapshot();
  })
})