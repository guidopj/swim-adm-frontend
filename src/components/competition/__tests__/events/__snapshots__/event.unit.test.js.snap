// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Events with state and props renders correctly 1`] = `
<div>
  <div
    className="MuiPaper-root MuiCard-root makeStyles-generalCard-5 MuiPaper-elevation1 MuiPaper-rounded"
  >
    <div
      className="MuiCardHeader-root"
    >
      <div
        className="MuiCardHeader-content"
      >
        <span
          className="MuiTypography-root MuiCardHeader-title MuiTypography-h5 MuiTypography-displayBlock"
        >
          Create New Event
        </span>
      </div>
    </div>
    <div
      className="MuiCardContent-root"
    >
      <div
        className="MuiGrid-root MuiGrid-container MuiGrid-spacing-xs-2"
      >
        <div
          className="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-12 MuiGrid-grid-md-6 MuiGrid-grid-lg-4"
        >
          <div
            className="MuiFormControl-root MuiTextField-root makeStyles-select-2"
          >
            <label
              className="MuiFormLabel-root MuiInputLabel-root MuiInputLabel-formControl MuiInputLabel-animated MuiInputLabel-shrink MuiFormLabel-filled"
              data-shrink={true}
              htmlFor="event_style"
              id="event_style-label"
            >
              meters
            </label>
            <div
              className="MuiInputBase-root MuiInput-root MuiInput-underline MuiInputBase-formControl MuiInput-formControl"
              onClick={[Function]}
            >
              <div
                aria-haspopup="listbox"
                aria-labelledby="event_style-label event_style"
                className="MuiSelect-root MuiSelect-select MuiSelect-selectMenu MuiInputBase-input MuiInput-input"
                id="event_style"
                onBlur={[Function]}
                onFocus={[Function]}
                onKeyDown={[Function]}
                onMouseDown={[Function]}
                role="button"
                tabIndex={0}
              >
                25
              </div>
              <input
                autoFocus={false}
                onAnimationStart={[Function]}
                type="hidden"
                value={25}
              />
              <svg
                aria-hidden={true}
                className="MuiSvgIcon-root MuiSelect-icon"
                focusable="false"
                viewBox="0 0 24 24"
              >
                <path
                  d="M7 10l5 5 5-5z"
                />
              </svg>
            </div>
          </div>
        </div>
        <div
          className="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-12 MuiGrid-grid-md-6 MuiGrid-grid-lg-4"
        >
          <div
            className="MuiFormControl-root MuiTextField-root makeStyles-select-2"
          >
            <label
              className="MuiFormLabel-root MuiInputLabel-root MuiInputLabel-formControl MuiInputLabel-animated"
              data-shrink={false}
              htmlFor="event_style"
              id="event_style-label"
            >
              style
            </label>
            <div
              className="MuiInputBase-root MuiInput-root MuiInput-underline MuiInputBase-formControl MuiInput-formControl"
              onClick={[Function]}
            >
              <div
                aria-haspopup="listbox"
                aria-labelledby="event_style-label event_style"
                className="MuiSelect-root MuiSelect-select MuiSelect-selectMenu MuiInputBase-input MuiInput-input"
                id="event_style"
                onBlur={[Function]}
                onFocus={[Function]}
                onKeyDown={[Function]}
                onMouseDown={[Function]}
                role="button"
                tabIndex={0}
              >
                <span
                  dangerouslySetInnerHTML={
                    Object {
                      "__html": "&#8203;",
                    }
                  }
                />
              </div>
              <input
                autoFocus={false}
                onAnimationStart={[Function]}
                type="hidden"
                value=""
              />
              <svg
                aria-hidden={true}
                className="MuiSvgIcon-root MuiSelect-icon"
                focusable="false"
                viewBox="0 0 24 24"
              >
                <path
                  d="M7 10l5 5 5-5z"
                />
              </svg>
            </div>
          </div>
        </div>
        <div
          className="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-12 MuiGrid-grid-md-6 MuiGrid-grid-lg-4"
        >
          <div
            className="MuiFormControl-root MuiTextField-root makeStyles-select-2"
          >
            <label
              className="MuiFormLabel-root MuiInputLabel-root MuiInputLabel-formControl MuiInputLabel-animated"
              data-shrink={false}
              htmlFor="genre"
              id="genre-label"
            >
              genre
            </label>
            <div
              className="MuiInputBase-root MuiInput-root MuiInput-underline MuiInputBase-formControl MuiInput-formControl"
              onClick={[Function]}
            >
              <div
                aria-haspopup="listbox"
                aria-labelledby="genre-label genre"
                className="MuiSelect-root MuiSelect-select MuiSelect-selectMenu MuiInputBase-input MuiInput-input"
                id="genre"
                onBlur={[Function]}
                onFocus={[Function]}
                onKeyDown={[Function]}
                onMouseDown={[Function]}
                role="button"
                tabIndex={0}
              >
                <span
                  dangerouslySetInnerHTML={
                    Object {
                      "__html": "&#8203;",
                    }
                  }
                />
              </div>
              <input
                autoFocus={false}
                onAnimationStart={[Function]}
                type="hidden"
                value=""
              />
              <svg
                aria-hidden={true}
                className="MuiSvgIcon-root MuiSelect-icon"
                focusable="false"
                viewBox="0 0 24 24"
              >
                <path
                  d="M7 10l5 5 5-5z"
                />
              </svg>
            </div>
          </div>
        </div>
        <div
          className="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-12 MuiGrid-grid-md-3 MuiGrid-grid-lg-4"
        >
          <div
            className="MuiFormControl-root MuiTextField-root makeStyles-select-2"
          >
            <label
              className="MuiFormLabel-root MuiInputLabel-root MuiInputLabel-formControl MuiInputLabel-animated MuiInputLabel-shrink MuiFormLabel-filled"
              data-shrink={true}
              htmlFor="age_from"
              id="age_from-label"
            >
              Age From
            </label>
            <div
              className="MuiInputBase-root MuiInput-root MuiInput-underline MuiInputBase-formControl MuiInput-formControl"
              onClick={[Function]}
            >
              <div
                aria-haspopup="listbox"
                aria-labelledby="age_from-label age_from"
                className="MuiSelect-root MuiSelect-select MuiSelect-selectMenu MuiInputBase-input MuiInput-input"
                id="age_from"
                onBlur={[Function]}
                onFocus={[Function]}
                onKeyDown={[Function]}
                onMouseDown={[Function]}
                role="button"
                tabIndex={0}
              >
                7
              </div>
              <input
                autoFocus={false}
                onAnimationStart={[Function]}
                type="hidden"
                value={7}
              />
              <svg
                aria-hidden={true}
                className="MuiSvgIcon-root MuiSelect-icon"
                focusable="false"
                viewBox="0 0 24 24"
              >
                <path
                  d="M7 10l5 5 5-5z"
                />
              </svg>
            </div>
          </div>
        </div>
        <div
          className="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-12 MuiGrid-grid-md-3 MuiGrid-grid-lg-4"
        >
          <div
            className="MuiFormControl-root MuiTextField-root makeStyles-select-2"
          >
            <label
              className="MuiFormLabel-root MuiInputLabel-root MuiInputLabel-formControl MuiInputLabel-animated MuiInputLabel-shrink MuiFormLabel-filled"
              data-shrink={true}
              htmlFor="age_to"
              id="age_to-label"
            >
              Age To
            </label>
            <div
              className="MuiInputBase-root MuiInput-root MuiInput-underline MuiInputBase-formControl MuiInput-formControl"
              onClick={[Function]}
            >
              <div
                aria-haspopup="listbox"
                aria-labelledby="age_to-label age_to"
                className="MuiSelect-root MuiSelect-select MuiSelect-selectMenu MuiInputBase-input MuiInput-input"
                id="age_to"
                onBlur={[Function]}
                onFocus={[Function]}
                onKeyDown={[Function]}
                onMouseDown={[Function]}
                role="button"
                tabIndex={0}
              >
                <span
                  dangerouslySetInnerHTML={
                    Object {
                      "__html": "&#8203;",
                    }
                  }
                />
              </div>
              <input
                autoFocus={false}
                onAnimationStart={[Function]}
                type="hidden"
                value={8}
              />
              <svg
                aria-hidden={true}
                className="MuiSvgIcon-root MuiSelect-icon"
                focusable="false"
                viewBox="0 0 24 24"
              >
                <path
                  d="M7 10l5 5 5-5z"
                />
              </svg>
            </div>
          </div>
        </div>
      </div>
    </div>
    <div
      className="MuiCardActions-root MuiCardActions-spacing"
    >
      <div
        className="MuiGrid-root makeStyles-actionButtons-4 MuiGrid-container MuiGrid-spacing-xs-4"
      >
        <div
          className="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-12 MuiGrid-grid-md-4"
        >
          <button
            className="MuiButtonBase-root MuiButton-root MuiButton-contained"
            disabled={false}
            onBlur={[Function]}
            onClick={[Function]}
            onDragLeave={[Function]}
            onFocus={[Function]}
            onKeyDown={[Function]}
            onKeyUp={[Function]}
            onMouseDown={[Function]}
            onMouseLeave={[Function]}
            onMouseUp={[Function]}
            onTouchEnd={[Function]}
            onTouchMove={[Function]}
            onTouchStart={[Function]}
            tabIndex={0}
            type="button"
          >
            <span
              className="MuiButton-label"
            >
              Create New Event
            </span>
          </button>
        </div>
        <div
          className="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-12 MuiGrid-grid-md-4"
        >
          <button
            className="MuiButtonBase-root MuiButton-root MuiButton-contained"
            disabled={false}
            onBlur={[Function]}
            onDragLeave={[Function]}
            onFocus={[Function]}
            onKeyDown={[Function]}
            onKeyUp={[Function]}
            onMouseDown={[Function]}
            onMouseLeave={[Function]}
            onMouseUp={[Function]}
            onTouchEnd={[Function]}
            onTouchMove={[Function]}
            onTouchStart={[Function]}
            tabIndex={0}
            type="button"
          >
            <span
              className="MuiButton-label"
            >
              Clean All Fields
            </span>
          </button>
        </div>
        <div
          className="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-12 MuiGrid-grid-md-4"
        >
          <a
            href="/"
            onClick={[Function]}
          >
            <button
              className="MuiButtonBase-root MuiButton-root MuiButton-contained"
              disabled={false}
              onBlur={[Function]}
              onDragLeave={[Function]}
              onFocus={[Function]}
              onKeyDown={[Function]}
              onKeyUp={[Function]}
              onMouseDown={[Function]}
              onMouseLeave={[Function]}
              onMouseUp={[Function]}
              onTouchEnd={[Function]}
              onTouchMove={[Function]}
              onTouchStart={[Function]}
              tabIndex={0}
              type="button"
            >
              <span
                className="MuiButton-label"
              >
                Go Back
              </span>
            </button>
          </a>
        </div>
      </div>
    </div>
  </div>
  <div
    className="MuiGrid-root makeStyles-tableContainer-6 MuiGrid-container"
  >
    <div
      className="MuiGrid-root MuiGrid-item MuiGrid-grid-md-10"
    >
      <div
        className="MuiPaper-root MuiTableContainer-root MuiPaper-elevation1 MuiPaper-rounded"
      >
        <h6
          className="MuiTypography-root MuiTypography-h6"
          id="tableTitle"
        >
          Events
        </h6>
        <table
          aria-label="simple table"
          className="MuiTable-root"
          role={null}
        >
          <thead
            className="MuiTableHead-root"
            role={null}
          >
            <tr
              className="MuiTableRow-root MuiTableRow-head"
              role={null}
            >
              <th
                aria-sort={null}
                className="MuiTableCell-root MuiTableCell-head"
                scope="col"
              >
                NRO
              </th>
              <th
                aria-sort={null}
                className="MuiTableCell-root MuiTableCell-head"
                scope="col"
              >
                METERS
              </th>
              <th
                aria-sort={null}
                className="MuiTableCell-root MuiTableCell-head"
                scope="col"
              >
                STYLE
              </th>
              <th
                aria-sort={null}
                className="MuiTableCell-root MuiTableCell-head"
                scope="col"
              >
                MIN AGE
              </th>
              <th
                aria-sort={null}
                className="MuiTableCell-root MuiTableCell-head"
                scope="col"
              >
                MAX AGE
              </th>
              <th
                aria-sort={null}
                className="MuiTableCell-root MuiTableCell-head"
                scope="col"
              >
                GENRE
              </th>
            </tr>
          </thead>
          <tbody
            className="MuiTableBody-root"
            role={null}
          >
            <tr
              className="MuiTableRow-root"
              role={null}
            >
              <th
                aria-sort={null}
                className="MuiTableCell-root MuiTableCell-body"
                role="cell"
                scope="row"
              >
                1
              </th>
              <th
                aria-sort={null}
                className="MuiTableCell-root MuiTableCell-body"
                role="cell"
                scope="row"
              >
                50
              </th>
              <th
                aria-sort={null}
                className="MuiTableCell-root MuiTableCell-body"
                role="cell"
                scope="row"
              >
                BACKSTROKE
              </th>
              <th
                aria-sort={null}
                className="MuiTableCell-root MuiTableCell-body"
                role="cell"
                scope="row"
              >
                18
              </th>
              <th
                aria-sort={null}
                className="MuiTableCell-root MuiTableCell-body"
                role="cell"
                scope="row"
              >
                23
              </th>
              <th
                aria-sort={null}
                className="MuiTableCell-root MuiTableCell-body"
                role="cell"
                scope="row"
              >
                MALE
              </th>
            </tr>
            <tr
              className="MuiTableRow-root"
              role={null}
            >
              <th
                aria-sort={null}
                className="MuiTableCell-root MuiTableCell-body"
                role="cell"
                scope="row"
              >
                2
              </th>
              <th
                aria-sort={null}
                className="MuiTableCell-root MuiTableCell-body"
                role="cell"
                scope="row"
              >
                50
              </th>
              <th
                aria-sort={null}
                className="MuiTableCell-root MuiTableCell-body"
                role="cell"
                scope="row"
              >
                BACKSTROKE
              </th>
              <th
                aria-sort={null}
                className="MuiTableCell-root MuiTableCell-body"
                role="cell"
                scope="row"
              >
                18
              </th>
              <th
                aria-sort={null}
                className="MuiTableCell-root MuiTableCell-body"
                role="cell"
                scope="row"
              >
                23
              </th>
              <th
                aria-sort={null}
                className="MuiTableCell-root MuiTableCell-body"
                role="cell"
                scope="row"
              >
                MALE
              </th>
            </tr>
            <tr
              className="MuiTableRow-root"
              role={null}
            >
              <th
                aria-sort={null}
                className="MuiTableCell-root MuiTableCell-body"
                role="cell"
                scope="row"
              >
                3
              </th>
              <th
                aria-sort={null}
                className="MuiTableCell-root MuiTableCell-body"
                role="cell"
                scope="row"
              >
                50
              </th>
              <th
                aria-sort={null}
                className="MuiTableCell-root MuiTableCell-body"
                role="cell"
                scope="row"
              >
                BACKSTROKE
              </th>
              <th
                aria-sort={null}
                className="MuiTableCell-root MuiTableCell-body"
                role="cell"
                scope="row"
              >
                18
              </th>
              <th
                aria-sort={null}
                className="MuiTableCell-root MuiTableCell-body"
                role="cell"
                scope="row"
              >
                23
              </th>
              <th
                aria-sort={null}
                className="MuiTableCell-root MuiTableCell-body"
                role="cell"
                scope="row"
              >
                MALE
              </th>
            </tr>
          </tbody>
        </table>
      </div>
    </div>
  </div>
</div>
`;
